### PR TITLE
Run github actions on all pushes / PRs

### PR DESCRIPTION
**Pull request by biobootloader**
Github actions are free since it's a public repo, this will help people catch things faster as @granawkins suggested

### CODE REVIEW

**Comment by biobootloader** [comment]
hmm annoying that now there are duplicated checks on PRs:
image

**Review by PCSwingle — APPROVED** [review]

File: .github/workflows/lint_and_test.yml [hunk 1]
```diff
@@ -3,13 +3,7 @@
...
 name: lint and test
 
-on:
-  push:
-    branches:
-      - main
-  pull_request:
-    branches:
-      - main
+on: [push, pull_request]
```
Comment (granawkins) [inline .github/workflows/lint_and_test.yml:6]:
```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
    inputs:
      branch:
        description: 'Branch to run on'
        required: true
        default: 'main'
```
Saw your comment about duplicating tests, that's not ideal.

What about instead allowing you to manually run the tests on a particular branch?

Comment (biobootloader) [reply .github/workflows/lint_and_test.yml:6]: I found an alternative approach here: https://wildwolf.name/github-actions-how-to-avoid-running-the-same-workflow-multiple-times/

**Comment by biobootloader** [comment]
I added these checks (found them here https://wildwolf.name/github-actions-how-to-avoid-running-the-same-workflow-multiple-times/). This should prevent the "pull-request" events from running unless the PR is from another repo. That should fix the duplicate checks being run on PRs that come from branches in this repo, but we'll see...

image

**Comment by biobootloader** [comment]
well I tried several ways to not have to duplicate the check on each job but to no avail

**Pull request merged**